src/shogun/base/SGObject.h

```diff
@@ -1344,30 +1373,15 @@ CSGObject* get_if_possible(const CSGObject* obj, std::string_view name, GetByNam
 	return result;
 }
 
-template<typename T>
-CSGObject* get_dispatch_all_base_types(const CSGObject* obj, std::string_view name,
-		T&& how)
-{
-	if (auto* result = get_if_possible<CKernel>(obj, name, how))
-		return result;
-	if (auto* result = get_if_possible<CFeatures>(obj, name, how))
-		return result;
-	if (auto* result = get_if_possible<CMachine>(obj, name, how))
-		return result;
-	if (auto* result = get_if_possible<CLabels>(obj, name, how))
-		return result;
-	if (auto* result = get_if_possible<CEvaluationResult>(obj, name, how))
-		return result;
-
-	return nullptr;
-}
-
 template<class T>
-CSGObject* get_by_tag(const CSGObject* obj, std::string_view name,
-		T&& how)
-{
-	return get_dispatch_all_base_types(obj, name, how);
-}
+CSGObject* get_by_tag(const CSGObject* obj, std::string_view name) {
```

Comment:
this whole thing could be deleted and go into a private function that is implemented in .cpp